### PR TITLE
Convert StarRating component to TypeScript

### PR DESCRIPTION
**Pull request by lavelle**

### CODE REVIEW

File: packages/thumbprint-react/components/StarRating/index.tsx [hunk 1]
```diff
@@ -14,13 +13,51 @@ const MAX_NUM_STARS = 5;
 // Smallest increment we render
 const PRECISION = 0.5;
 
-const StarIcon = p => (
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const StarIcon = (p: any): JSX.Element => (
```
Comment (danoc) [inline packages/thumbprint-react/components/StarRating/index.tsx]: Does `React.SVGAttributes` or `React.SVGProps` work?

File: packages/thumbprint-react/components/StarRating/index.tsx [hunk 2]
```diff
@@ -14,13 +13,51 @@ const MAX_NUM_STARS = 5;
...
+interface PropTypes {
+    /**
+     * Number from 0-5 at increments of 0.5. Numbers between these steps will be rounded.
+     */
+    rating: number;
+    /**
+     * Number from 0-5 at increments of 1. `hoverRating` trumps `rating` with respect to star
+     * highlighting.
+     */
+    hoverRating?: number;
```
Comment (danoc) [inline packages/thumbprint-react/components/StarRating/index.tsx]:
```suggestion
    hoverRating?: 0 | 1 | 2 | 3 | 4 | 5;
```

File: packages/thumbprint-react/components/StarRating/index.tsx (hunk 1 above)
Comment (lavelle) [reply packages/thumbprint-react/components/StarRating/index.tsx]: done. good idea

File: packages/thumbprint-react/components/StarRating/index.tsx (hunk 2 above)
Comment (lavelle) [reply packages/thumbprint-react/components/StarRating/index.tsx]: done

**Pull request merged**